internal/config/config.go

```diff
@@ -1,63 +1,74 @@
-	SeedURL string `env:"SEED_URL" envDefault:"https://raw.githubusercontent.com/cosmos/chain-registry/master/akash/chain.json"`
+type ServerConfig struct {
+	Listen     string        `mapstructure:"listen"`
+	ListenGRPC string        `mapstructure:"listen-grpc"`
+	Timeouts   TimeoutConfig `mapstructure:"timeouts"`
+}
 
-	// How frequently fetch SEED_URL for updates.
-	SeedRefreshInterval time.Duration `env:"SEED_REFRESH_INTERVAL" envDefault:"5m"`
+type TimeoutConfig struct {
+	Read  time.Duration `mapstructure:"read"`
+	Write time.Duration `mapstructure:"write"`
+	Idle  time.Duration `mapstructure:"idle"`
+}
 
-	// Expected chain ID.
-	ChainID string `env:"CHAIN_ID" envDefault:"akashnet-2"`
+type TLSConfig struct {
+	Autocert AutocertConfig `mapstructure:"autocert"`
+	Cert     string         `mapstructure:"cert"`
+	Key      string         `mapstructure:"key"`
+}
 
-	// How slow on average a node needs to be to be marked as unhealthy.
-	HealthyThreshold time.Duration `env:"HEALTHY_THRESHOLD" envDefault:"10s"`
+type AutocertConfig struct {
+	Email string   `mapstructure:"email"`
+	Hosts []string `mapstructure:"hosts"`
+}
 
-	// Percentage of request errors deemed acceptable.
-	HealthyErrorRateThreshold float64 `env:"HEALTHY_ERROR_RATE_THRESHOLD" envDefault:"30"`
+type SeedConfig struct {
+	URL             string        `mapstructure:"url"`
+	RefreshInterval time.Duration `mapstructure:"refresh-interval"`
+	ChainID         string        `mapstructure:"chain-id"`
+	AdditionalNodes struct {
+		RPC  []string `mapstructure:"rpc"`
+		REST []string `mapstructure:"rest"`
+		GRPC []string `mapstructure:"grpc"`
+	} `mapstructure:"additional-nodes"`
+}
 
-	// How long in the past requests are considered to check for status codes.
-	HealthyErrorRateBucketTimeout time.Duration `env:"HEALTHY_ERROR_RATE_BUCKET_TIMEOUT" envDefault:"1m"`
+type HealthConfig struct {
+	HealthyThreshold    time.Duration `mapstructure:"healthy-threshold"`
+	ProxyRequestTimeout time.Duration `mapstructure:"proxy-request-timeout"`
+}
 
-	// Request timeout for a proxied request.
-	ProxyRequestTimeout time.Duration `env:"PROXY_REQUEST_TIMEOUT" envDefault:"15s"`
+type CORSConfig struct {
+	AllowOrigin  string `mapstructure:"allow-origin"`
+	AllowMethods string `mapstructure:"allow-methods"`
+	AllowHeaders string `mapstructure:"allow-headers"`
+}
 
-	// How much chance (in %, 0-100), a node marked as unhealthy have to get a
-	// request again and recover.
-	UnhealthyServerRecoverChancePct int `env:"UNHEALTHY_SERVER_RECOVERY_CHANCE_PERCENT" envDefault:"1"`
+type Config struct {
+	Server ServerConfig `mapstructure:"server"`
+	TLS    TLSConfig    `mapstructure:"tls"`
+	Seed   SeedConfig   `mapstructure:"seed"`
+	Health HealthConfig `mapstructure:"health"`
+	CORS   CORSConfig   `mapstructure:"cors"`
 }
 
-func Must() Config {
-	cfg, err := env.ParseAsWithOptions[Config](env.Options{
-		Prefix: "AKASH_PROXY_",
-	})
+// Must returns the current configuration or panics if it cannot be loaded.
```

Comment:
Lets rename it as it is longer `Must`
also `if` statement can be compacted to
```
 	if err := v.Unmarshal(&cfg); err != nil {
 		return Config{}
 	}
```

Reply:
It's still called `Must` (`func Must(v *viper.Viper) Config {`).
I compacted the if statement. 👍🏻